hawc/apps/lit/api.py

```diff
@@ -166,6 +169,64 @@ def tag_heatmap(self, request, pk):
         export = FlatExport(df=df, filename=f"df-{instance.id}")
         return Response(export)
 
+    @transaction.atomic()
+    @action(
+        detail=True, methods=("patch",), url_path="replace-hero",
+    )
+    def replace_hero(self, request, pk):
+        assessment = self.get_object()
+        body = json.loads(request.body)
+        replace = body.get("replace", list())
+        replace_unzipped = list(zip(*replace))
+        ref_ids = replace_unzipped[0]
+        hero_ids = replace_unzipped[1]
+
+        # make sure all references are HERO and in assessment
+        matching_references = assessment.references.filter(
+            id__in=ref_ids, identifiers__database=constants.HERO
+        )
+        if matching_references.count() != len(ref_ids):
+            raise exceptions.ValidationError("All references must be from selected assessment.")
+
+        # set hero ref
+        for index, (ref, hero) in enumerate(replace):
+            reference = models.Reference.objects.get(id=ref)
+            hero_identifier = reference.identifiers.get(database=constants.HERO)
+            setattr(hero_identifier, "unique_id", str(hero))
+            hero_identifier.save()
+
+        # update content
+        tasks.update_hero_content.apply(args=[hero_ids])
+
+        # update fields with content
+        for ref in ref_ids:
```

Comment:
so this will work locally, but will fail in prod. The reason is `update_hero_content` is async and will not finish before you're updating the  content.  Locally, we run celery tasks in the same thread, so that's why it looks ok here.

To do this correctly, you'll need convert this to another celery task, and then chain them together: https://docs.celeryproject.org/en/stable/userguide/canvas.html#chains

Reply:
I've done it once before here: https://github.com/NIEHS/orio-web/blob/master/project/analysis/tasks.py#L48